core/src/main/java/com/scalar/db/common/error/CoreError.java

```diff
@@ -690,6 +690,42 @@ public enum CoreError implements ScalarDbError {
       ""),
   DATA_LOADER_ERROR_METHOD_NULL_ARGUMENT(
       Category.USER_ERROR, "0151", "Method null argument not allowed", "", ""),
+  OUT_OF_RANGE_COLUMN_VALUE_FOR_DATE(
+      Category.USER_ERROR,
+      "0152",
+      "This DATE column value is out of the valid range. It must be between 1000-01-01 and 9999-12-12. Value: %s",
+      "",
+      ""),
+  SUBMICROSECOND_PRECISION_NOT_SUPPORTED_FOR_TIME(
+      Category.USER_ERROR,
+      "0153",
+      "This TIME column value precision cannot be smaller than one microsecond. Value: %s",
```

Comment:
I just realized that, since we're talking about time, this should be `shorter` instead of `smaller`.

```suggestion
      "This TIME column value precision cannot be shorter than one microsecond. Value: %s",
```